xbmc/pvr/PVRDatabase.cpp

```diff
@@ -624,18 +608,22 @@ bool CPVRDatabase::Persist(CPVRChannel &channel)
         channel.IconPath().c_str(), channel.ChannelName().c_str(), 0, (channel.EPGEnabled() ? 1 : 0), channel.EPGScraper().c_str(), channel.LastWatched(), channel.ClientID(),
         channel.EpgID());
 
-    QueueInsertQuery(strQuery);
+    ExecuteQuery(strQuery);
+
+    /* Update the channel ID */
+    int newChannelId = GetChannelId(channel.ClientID(), channel.UniqueID());
+    channel.SetChannelID(newChannelId);
```

Comment:
you don't need to query the db again to get the id of the row being inserted. should be replaced by `m_pDS->lastinsertid()`. don't forget to cast to int.